package-system/zlib/Findzlib.cmake.template

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# this file actually ingests the library and defines targets.
+set(TARGET_WITH_NAMESPACE "3rdParty::zlib")
+if (TARGET $${TARGET_WITH_NAMESPACE})
+    return()
+endif()
+
+set(zlib_INCLUDE_DIR $${CMAKE_CURRENT_LIST_DIR}/zlib/include)
+set(zlib_LIBS_DIR $${CMAKE_CURRENT_LIST_DIR}/zlib/lib)
+
+if ($${PAL_PLATFORM_NAME} STREQUAL "Linux")
```

Comment:
are you using this to escape it for when it is generated? a better way is to wrap the variables you want to replace with `@` and then generating with `@ONLY`: https://cmake.org/cmake/help/latest/command/string.html#configure 
That way is easier to distinguish the variables that will be expanded vs the ones that will stay

Reply:
These template files are processed by Python, not cmake, that is why they are escaped like this.

Reply:
thanks for the clarification

Reply:
I'd like to add a feature to pull_and_build_from_git that lets you just use a cmake find file as is.  I'm not sure it actually supports just not using templates.

Reply:
Sounds good to me